cmake/Platform/Windows/Packaging/Tests/test_installer.py

```diff
@@ -0,0 +1,217 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Pytest test configuration file.
+
+Example command to run these tests:
+
+pytest cmake/Platform/Windows/Packaging/Tests \
+    --capture=no \
+    --log-file=C:/workspace/test_installer.log \
+    --installer-uri=file:///C:/path/to/o3de_installer.exe \
+    --install-root=C:/O3DE/0.0.0.0 \
+    --project-path=C:/workspace/TestProject
+
+Alternately, the installer-uri can be an s3 or web URL. For example:
+--installer-uri=s3://bucketname/o3de_installer.exe
+--installer-uri=https://o3de.binaries.org/o3de_installer.exe
+
+"""
+import pytest
+import json
+from pathlib import Path
+from subprocess import TimeoutExpired
+
+
+@pytest.fixture(scope="session")
+def test_installer_fixture(context):
+    """ Installer executable succeeds. """
+    assert context.installer_path.is_file(), f"Invalid installer path {context.installer_path}"
+
+    # when the installer is run with timeout of 30 minutes
+    result = context.run([str(context.installer_path) , f"InstallFolder={context.install_root}", "/quiet"], timeout=30*60)
+
+    # the installer succeeds
+    assert result.returncode == 0, f"Installer failed with exit code {result.returncode}"
+
+    # the install root is created
+    assert context.install_root.is_dir(), f"Invalid install root {context.install_root}"
+
+
+@pytest.mark.parametrize(
+    "filename", [
+        pytest.param('Editor.exe'),
+        pytest.param('AssetProcessor.exe'),
+        pytest.param('MaterialEditor.exe'),
+        pytest.param('o3de.exe'),
+    ]
+)
+def test_binaries_exist(test_installer_fixture, context, filename):
+    """ Installer key binaries exist. """
+    assert (context.engine_bin_path / filename).is_file(), f"{filename} not found in {context.engine_bin_path}"
+
+
+@pytest.fixture(scope="session")
+def test_o3de_registers_engine_fixture(test_installer_fixture, context):
+    """ Engine is registered when o3de.exe is run. """
+    try:
+        result = context.run([str(context.engine_bin_path / 'o3de.exe')], timeout=15)
+
+        # o3de.exe should not close with an error code 
+        assert result.returncode == 0, f"o3de.exe failed with exit code {result.returncode}"
+    except TimeoutExpired as e:
+        # we expect to close the app ourselves
+        pass
+
+    engine_json_path = context.install_root / 'engine.json'
+    with engine_json_path.open('r') as f:
+        engine_json_data = json.load(f)
+
+    engine_name = engine_json_data['engine_name']
+
+    manifest_path = context.home_path / '.o3de' / 'o3de_manifest.json'
+    with manifest_path.open('r') as f:
+        manifest_json_data = json.load(f)
```

Comment:
There is a [manifest.get_engine_json_data](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/manifest.py#L479-L488) function that loads the engine.json file and retrieves it's json data

You can then just invoke [manifest.get_registered](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/manifest.py#L561-L568)  with the `engine_name=engine_name` parameter to a path to the registered engine if it exist.

Reply:
Hmm for some reason get_registered() is failing on Jenkins, looking into it now.

Reply:
Ugh...
We can use the `USERPROFILE` environment variable if the sys.platform is "windows'
https://docs.python.org/3/library/os.path.html#os.path.expanduser